### PR TITLE
chore: add postinstall deprecation warning to old packages

### LINKED ISSUE #12495: Add deprecation warning in terminal for carbon-components and carbon-components-react
As part of [the plan to deprecate these packages](https://github.com/carbon-design-system/carbon/discussions/12179), the following items need to happen:

### Tasks
- [ ] Add a postinstall message to carbon-components and carbon-components-react in the terminal stating (separately) our recommendation to move to @carbon/styles and @carbon/react, respectively.
- [ ] In the message, link back to the discussion issue https://github.com/carbon-design-system/carbon/discussions/12179 for information on the package strategy, deprecation, and plan of action.

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #12495 

Adds deprecation warnings to `carbon-components` and `carbon-components-react`

#### Changelog

**New**

- packages/carbon-components-react/scripts/postinstall.js: script to log deprecation warning message
- packages/carbon-components/scripts/postinstall.js: script to log deprecation warning message

**Changed**

- packages/carbon-components-react/package.json: call `postinstall` script, add `chalk` dependency
- packages/carbon-components/package.json:  call `postinstall` script, add `chalk` dependency

#### Testing / Reviewing

You can test manually by running `yarn run postinstall`  inside the packages dirs: `packages/carbon-components-react` and `packages/carbon-components`
image

### CODE REVIEW

**Review by aledavila — APPROVED** [review]
lgtm. can see it when running. just some merge conflicts

_thats the longest branch name i've ever seen 😆_

File: packages/carbon-components-react/scripts/postinstall.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const chalk = require('chalk');
+
+console.warn(
+  chalk.yellow('Warning: ') +
+    chalk.cyan('carbon-components-react') +
+    " is soon to be deprecated and is no longer receiving updates, we recommend to move to it's replacement, " +
```
Comment (tay1orjones) [inline packages/carbon-components-react/scripts/postinstall.js]:
```suggestion
    " will soon be deprecated and no longer receive updates. We recommend to move to it's replacement, " +
```

File: packages/carbon-components/scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const chalk = require('chalk');
+
+console.warn(
+  chalk.yellow('Warning: ') +
+    chalk.cyan('carbon-components') +
+    " is soon to be deprecated and is no longer receiving updates, we recommend to move to it's replacement, " +
```
Comment (tay1orjones) [inline packages/carbon-components/scripts/postinstall.js]:
```suggestion
    " will soon be deprecated and no longer receive updates. We recommend to move to it's replacement, " +
```

**Review by tay1orjones — CHANGES_REQUESTED** [review]
We'll still be updating these packages while we're on v11.x, so just a couple suggestions on the phrasing of the message to reflect that.

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**